### PR TITLE
[FIX]: Import wallets for an existing key

### PR DESCRIPTION
**Pull request by rastajpa**

### CODE REVIEW

File: src/providers/key/key.ts [hunk 1]
```diff
@@ -285,4 +285,8 @@ export class KeyProvider {
       return key1.fingerPrint === key2.fingerPrint;
     else return key1.id === key2.id;
   }
+
+  public getKeyByFingerprint(key) {
```
Comment (matiu) [inline src/providers/key/key.ts:289]: Given "isMatch" will check first fingerPrint and then id, 
maybe a better will be "getMatchedKey"

File: src/providers/profile/profile.ts [hunk 2]
```diff
@@ -926,6 +926,14 @@ export class ProfileProvider {
     this.logger.info('Importing Wallet xPrivKey');
     opts.xPrivKey = xPrivKey;
     return this.serverAssistedImport(opts).then(data => {
+      const key = this.keyProvider.getKeyByFingerprint(data.key);
+      if (key) {
```
Comment (matiu) [inline src/providers/profile/profile.ts:930]: a comment would help help. // If we have the key already, bind the new wallets to the current key.

**Comment by matiu** [comment]
LGTM, left some minor comments.

File: src/providers/key/key.ts (hunk 1 above)
Comment (rastajpa) [reply src/providers/key/key.ts:289]: getMatchedKey looks better

**Pull request merged**